core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -929,6 +929,29 @@ public static String charFromUtf8(int n) {
     return String.valueOf(Character.toChars(n));
   }
 
+  /**
+   * SQL CODE_POINTS_TO_BYTES function.
+   */
+  public static @Nullable ByteString codePointsToBytes(List codePoints) {
```

Comment:
Is `codePoints` always a list of integers? If so you could set the parameter type as List<Integer> and then some of the logic below becomes unnecessary. Let me know if you see problems with this.

Reply:
Thanks for you review:) @tanclary.
Actually, the elements of codePoints can be of type Long or Integer. Additionally, let's consider an input of `17179869185` (which is equal to `(1L << 34) + 1L`). Although this is an illegal argument, when it is cast to an integer (using the `intValue()` method), it will become 1! This will result in an incorrect output. So, I believe using this logic is reasonable. WDYT?

Reply:
I notice that operand type chceker is `ARRAY_OF_INTEGER`, so IMO in runtime, it always be array of integer as operand, and not necessary to add the check logic for long value.

Reply:
@herunkang2018 Thanks for your advice. Calcite often use `SqlTypeFamily.INTEGER`(which is collection of `INT_TYPES` in `SqlTypeName`) as a integer operand checker instead of `SqlTypeName.INTEGER`. This can easily be confusing. So `ARRAY_OF_INTEGER` means an array whose component could be `TINYINT, SMALLINT, INTEGER or BIGINT`.
Back to your comment, array component could be any of the `INT_TYPES`, including `BIGINT`(Correspond Long in java). The reason I allow the `BIGINT` component to pass this checker is because if `BIGINT` component is invaild input type, `code_points_to_byte(array[cast(1 as bigint)])` will fail during operand type checking, this does not meet expectations. And  if I execute `select code_points_to_bytes(array[2147483648])` in BigQuery, BigQuery will throw a exception :`Input arguments of CODE_POINTS_TO_BYTES out of range: 2147483648` rather than input type error(Actully BigQuery see all integers as int64).

Reply:
Thanks for clarification, I got it.